### ISSUE TITLE
[BUG] External Users can deleted federated channels

### ISSUE THREAD

**Issue by ArchdukeTim** [RocketChat/Rocket.Chat#20209]
### Description:

A user that has permission to delete channels in server A can be invited to and subsequently delete a channel in server B. 

### Steps to reproduce:

1. Federate two servers, A and B
2. Invite user X from server A into a channel J in server B
3. User X can delete J in server B using his permissions in server A

### Expected behavior:

External users cannot delete federated channels
### Actual behavior:

External users can delete federated channels.

### Server Setup Information:

- Version of Rocket.Chat Server: 3.9.4
- Operating System: linux x64
- Deployment Method: tar
- Number of Running Instances: 2
- DB Replicaset Oplog: 
- NodeJS Version: 12.18.4
- MongoDB Version: 4.4.2

### Client Setup Information

- Desktop App or Browser Version: Desktop
- Operating System: macOS

**Comment by KevLehman** [comment]
Hey, thanks for raising up the issue. Quick question: is this a duplicate of #20467 ?

**Comment by ArchdukeTim** [comment]
Yes it is. I got an error submitting that one and couldn't find it again. Closing that one now.
